components/slideshow.tsx

```diff
@@ -0,0 +1,94 @@
+'use client';
+import Image from 'next/image';
+import { useCallback, useEffect, useState } from 'react';
+import { FaChevronLeft, FaChevronRight } from 'react-icons/fa';
+
+export default function Slideshow({ images }: { images: string[] }) {
+  const [currentSlide, setCurrentSlide] = useState<number>(0);
+  const [hovered, setHovered] = useState<boolean>(false);
+
+  const prevSlide = useCallback(() => {
+    setCurrentSlide((prev: number) =>
+      prev === 0 ? images.length - 1 : prev - 1
+    );
+  }, [images.length]);
+
+  const nextSlide = useCallback(() => {
+    setCurrentSlide((prev: number) =>
+      prev === images.length - 1 ? 0 : prev + 1
+    );
+  }, [images.length]);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      nextSlide();
+    }, 3000);
+
+    return () => clearInterval(intervalId);
+  }, [nextSlide]);
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (hovered) {
+        if (event.key === 'ArrowLeft') {
+          prevSlide();
+        } else if (event.key === 'ArrowRight') {
+          nextSlide();
+        }
+      }
+    };
+
+    window.addEventListener('keydown', handleKeyDown);
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [hovered, prevSlide, nextSlide]);
+
+  return (
+    <div
+      className="relative overflow-hidden"
+      onMouseEnter={() => setHovered(true)}
+      onMouseLeave={() => setHovered(false)}
+    >
+      <button
+        className={`prev absolute top-1/2 z-10 -translate-y-1/2  transform opacity-0 transition-opacity duration-300 hover:opacity-100`}
```

Comment:
This is wrong, you're not using the `group-*` syntax which dodges the whole point. Either way, why are you trying to hide the buttons when not hovering on the slideshow anyway? It's not intuitive. Make it `opacity-60` for not hovered and `opacity-100` for hovered. Leave the group thing.

Also, what does the `prev` class do here?

Reply:
Oh I see, so that way the buttons will be in focus the moment we hover the slideshow and not the buttons themselves.